### DISCUSSION TITLE
client-(dis)connect scripts

### DISCUSSION THREAD

**Discussion by lx1** [schwabe/ics-openvpn#1397] [Q&A]
Hi all. First of, big thanks for OVPN for Android.

I just wonder if client-connect and client-disconnect options are supported in OpenVPN for Android. I found no doc on this topic. 

Background:
I'd like to have them to be able to reconnect the Android' SIP client when the vpn starts. Too bad, when I leave my work LAN and start the VPN with it, for some reason the Android native SIP does not reconnect automatically. It seems to continue to try to reach it locally and I have to restart it (particularly, reset the feature "receive incoming calls).
I can script this option reset, and would like to call it upon VPN connect and disconnect.

If the feature is supported, it would be helpful to have an example where to put the scripts in Android, just not to overwrite them with phone rom updates, and how to log messages.

Thank you, Luigi

**Comment by schwabe** [comment]
They are not supported as it is extremely hard to get them working and they are of little use otherwise. Technically OpenVPN will try to execute the scripts if you add the options as custom options but as far as I can recall it is very hard to get scripts working without tinkering. The more modern Android version might block calling executables/script that are not from the apk outright.

An easier way is to listen to the network event or use the AIDL API of my app to get event of the VPN.

**Comment by Naimjonsky** [comment]
Good day to all.  
I want to thank for this app and of course for providing the source code.  I have almost the same question.  For example, I have an application and I want to connect, disconnect and check the connection status.  
Since I am not an android developer, I would be very happy to see examples of these functions.

Thank you.
